input/parser_test.go

```diff
@@ -1,105 +1,209 @@
 package input
 
-import "testing"
+import (
+	"bytes"
+	"reflect"
+	"testing"
+)
 
-/*
- * RFC5424 parser tests
- */
+type ParserTestCases []ParserTestCase
 
-func Test_SuccessfulRFC5424Parsing(t *testing.T) {
-	p := NewRFC5424Parser()
+type ParserTestCase struct {
+	fmt     string
+	fail    bool
+	tests   []string
+	results []map[string]interface{}
+}
 
-	tests := []struct {
-		message  string
-		expected RFC5424Message
-	}{
-		{
-			message:  "<134>1 2003-08-24T05:14:15.000003-07:00 ubuntu sshd 1999 - password accepted",
-			expected: RFC5424Message{Priority: 134, Version: 1, Timestamp: "2003-08-24T05:14:15.000003-07:00", Host: "ubuntu", App: "sshd", Pid: 1999, MsgId: "-", Message: "password accepted"},
-		},
-		{
-			message:  "<33>5 1985-04-12T23:20:50.52Z test.com cron 304 - password accepted",
-			expected: RFC5424Message{Priority: 33, Version: 5, Timestamp: "1985-04-12T23:20:50.52Z", Host: "test.com", App: "cron", Pid: 304, MsgId: "-", Message: "password accepted"},
-		},
-		{
-			message:  "<1>0 1985-04-12T19:20:50.52-04:00 test.com cron 65535 - password accepted",
-			expected: RFC5424Message{Priority: 1, Version: 0, Timestamp: "1985-04-12T19:20:50.52-04:00", Host: "test.com", App: "cron", Pid: 65535, MsgId: "-", Message: "password accepted"},
-		},
-		{
-			message:  "<1>0 2003-10-11T22:14:15.003Z test.com cron 65535 msgid1234 password accepted",
-			expected: RFC5424Message{Priority: 1, Version: 0, Timestamp: "2003-10-11T22:14:15.003Z", Host: "test.com", App: "cron", Pid: 65535, MsgId: "msgid1234", Message: "password accepted"},
-		},
-		{
-			message:  "<1>0 2003-08-24T05:14:15.000003-07:00 test.com cron 65535 - JVM NPE\nsome_file.java:48\n\tsome_other_file.java:902",
-			expected: RFC5424Message{Priority: 1, Version: 0, Timestamp: "2003-08-24T05:14:15.000003-07:00", Host: "test.com", App: "cron", Pid: 65535, MsgId: "-", Message: "JVM NPE\nsome_file.java:48\n\tsome_other_file.java:902"},
+var parserTests = ParserTestCases{
+	ParserTestCase{
+		fmt: "syslog",
+		tests: []string{
```

Comment:
These tests are too difficult to maintain now, with the "input" and the "expected" disconnected like this. The code must use the older style.